PostHog/PostHogSDK.swift

```diff
@@ -490,6 +490,13 @@ let maxRetryDelay = 30.0
             if shouldReloadFlagsForTesting {
                 reloadFeatureFlags()
             }
+        } else if !(userProperties?.isEmpty ?? true) || !(userPropertiesSetOnce?.isEmpty ?? true) {
```

Comment:
I just noticed we need an extra check here.
[PostHog/posthog-android@`ef2e552` (#205)](https://github.com/PostHog/posthog-android/pull/205/commits/ef2e552e9318ba9c4a0ae17bdfe8789986871578)
See that the $set event should only happen if its for the very same distinctId

Reply:
The JS SDK instead of not calling capture(distinctId, event=$set) which is what we do here, doesn't use the given distinctId and uses the cached distinctId, I find that odd.

identify(userA, {a=b})
identify(userB, {a=c})

in our case, it won't work since the user is identified as userA, and they are trying to update user props from the userB.
In JS, they will update the user props {a=c} for userA.

So I think my approach is more defensive to user errors.